### DISCUSSION TITLE
Docker compose flexibility

### DISCUSSION THREAD

**Discussion by bjoernwuest** [immich-app/immich#13315] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Would love to see to be more flexible with the docker compose, i.e.:

* postgres must have service name "database" and used postgres is a special version with pgvecto extension. People (like me) who already have a compose environment with a "database" service run into collissions, especially when to maintain only one compose entry point. So please ad a "DB_HOST" and "DB_PORT" environment variable where necessary
* Same applies to redis :)

Mentioned in other discussions is the use of password files, which can be (without using Swarm) realised with construct like:
```
    environment:
      - POSTGRES_PASSWORD_FILE=/postgres_password.secret
    volumes:
      - /mylocalfileystem/postgres.secret:/postgres_password.secret:ro
```

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
1. You're free to edit the compose as you please
2. Compose stacks are isolated (unless you specifically choose to break that isolation), so having a `database` in multiple different composes is fine.
3. These env vars already exist.
4. Secret files are already supported
